consensus/core/src/commit_finalizer.rs

```diff
@@ -113,15 +113,18 @@ impl CommitFinalizer {
 
     async fn run(mut self, mut receiver: UnboundedReceiver<CommittedSubDag>) {
         while let Some(committed_sub_dag) = receiver.recv().await {
-            let finalized_commits = if self.context.protocol_config.mysticeti_fastpath() {
+            let already_finalized = !self.context.protocol_config.mysticeti_fastpath()
+                || committed_sub_dag.recovered_rejected_transactions;
+            let finalized_commits = if !already_finalized {
                 self.process_commit(committed_sub_dag).await
             } else {
                 vec![committed_sub_dag]
             };
             if !finalized_commits.is_empty() {
+                self.try_update_gc_round(finalized_commits.last().unwrap().leader.round);
```

Comment:
Why is it ok to do the GC here instead of later?

Reply:
Actually looking this again I understand it's necessary to ensure that no stale vote information will be read?